### PR TITLE
Added functions merge_with and volume to BoundingBox

### PR DESCRIPTION
**Pull request by GivAlz**
Adding two functions to the BoundingBox Class: I will need them soon.

I hope this time everything is ok!

Thank you,
Giovanni

### CODE REVIEW

**Comment by GivAlz** [comment]
I shall close the pull request as I want to add one more function to the same class.

File: tests/base/bounding_box_2.cc [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+
+// test for BoundingBox<unsigned int spacedim> which test basic stuff:
+// creation in various dimensions and checking for points inside
```
Comment (tjhei) [inline tests/base/bounding_box_2.cc]: please update this comment

File: source/base/bounding_box.cc [hunk 2]
```diff
@@ -32,10 +32,31 @@ bool BoundingBox<spacedim,Number>::point_inside (const Point<spacedim, Number> &
 }
 
 template <int spacedim, typename Number>
+void BoundingBox<spacedim,Number>::merge_with(const BoundingBox<spacedim,Number> &BBox)
+{
+  for (unsigned int i=0; i < spacedim; ++i)
+    {
+      if ( BBox.boundary_points.first[i] < this->boundary_points.first[i] )
```
Comment (tjhei) [inline source/base/bounding_box.cc]: any reason you are not using ``std::min``?

File: include/deal.II/base/bounding_box.h [hunk 3]
```diff
@@ -77,15 +77,25 @@ class BoundingBox
   const std::pair<Point<spacedim,Number>,Point<spacedim,Number>> &get_boundary_points () const;
 
   /**
+   * Merging function
```
Comment (tjhei) [inline include/deal.II/base/bounding_box.h]: Please extend this comment so say something like "grow the current object to also contain @p other"

File: include/deal.II/base/bounding_box.h [hunk 4]
```diff
@@ -77,15 +77,25 @@ class BoundingBox
...
+   */
+  void merge_with(const BoundingBox<spacedim,Number> &);
```
Comment (tjhei) [inline include/deal.II/base/bounding_box.h]: please name the argument (for example ``other``)

File: source/base/bounding_box.cc [hunk 5]
```diff
@@ -32,10 +32,31 @@ bool BoundingBox<spacedim,Number>::point_inside (const Point<spacedim, Number> &
 }
 
 template <int spacedim, typename Number>
+void BoundingBox<spacedim,Number>::merge_with(const BoundingBox<spacedim,Number> &BBox)
```
Comment (tjhei) [inline source/base/bounding_box.cc]: we typically use lower case variable names.

**Comment by GivAlz** [comment]
Thank you very much for the comments: there's really a lot I have to learn so thank you for the patience in correcting everything line by line!!

**Comment by bangerth** [comment]
It's going to get easier as you learn the kind of style we like to see. In the meantime, thanks for your good attitude with all of the changes people request - we're glad to have you in the project!

File: include/deal.II/base/bounding_box.h [hunk 6]
```diff
@@ -77,15 +77,27 @@ class BoundingBox
   const std::pair<Point<spacedim,Number>,Point<spacedim,Number>> &get_boundary_points () const;
 
   /**
+   * Merging function: enlarges the current object so that it contains @p other_bbox .
```
Comment (davydden) [inline include/deal.II/base/bounding_box.h]: maybe: ~Merging function:~ Enlarge the...

**Pull request merged**